frontend/src/components/PsyListing/PsyListing.jsx

```diff
@@ -69,18 +87,46 @@ const PsyListing = () => {
     }
   };
 
+  const success = pos => {
+    const { longitude, latitude } = pos.coords;
+    setCoords({ longitude, latitude });
+    setGeoStatus(geoStatusEnum.GRANTED);
+  };
+
+  const errors = () => {
+    setGeoStatus(geoStatusEnum.DENIED);
+  };
+
+  const getGeolocation = state => {
+    if (state === 'granted') {
+      navigator.geolocation.getCurrentPosition(success);
+    } else if (state === 'prompt') {
+      navigator.geolocation.getCurrentPosition(success, errors);
+    } else if (state === 'denied') {
+      setGeoStatus(geoStatusEnum.DENIED);
+    }
+  };
+
+  const checkGeolocationPermission = () => {
+    if (!coords) {
```

Comment:
Why don't we always do it ?

Reply:
Performance : avoid getting position if we already have it.
Can be discussed.

Reply:
can be a pain if the first location was not really precise
a don't really know how it works :/